### PR TITLE
Move exporting

### PR DESCRIPTION
**Pull request by drets**
Problem: https://github.com/heutagogy/heutagogy-frontend/issues/33

Solution:
• move export to modal window
• hide checkboxes in main table since they're unused

### CODE REVIEW

**Comment by rasendubi** [comment]
Travis is experiencing issues due to AWS S3 outage.

Tested everything locally. Works fine. :+1:

File: src/components/ExportModal/ExportModal.js [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+    this.state = { selectedRows: [] };
+  }
+
+  componentWillMount() {
+    this.props.loadEntities({
+      href: '/bookmarks?per_page=100',
+      type: ARTICLES_VIEW_STATE,
+      schema: arrayOf(articleSchema),
+    }).then(() => {
+      this.loadAllArticlesFromServer(this.props.linkHeader.get('last'));
```
Comment (rasendubi) [inline src/components/ExportModal/ExportModal.js:60]: Can we not fetch all articles from the server but reuse ones in redux store?

File: src/components/ExportModal/ExportModal.js [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        label={'Save'}
+        primary
+        style={inlineStyles.submit}
+        onTouchTap={this.handleSubmit}
+      />,
+      <FlatButton
+        label={'Cancel'}
+        primary
+        style={inlineStyles.close}
+        onTouchTap={this.handleClose}
```
Comment (rasendubi) [inline src/components/ExportModal/ExportModal.js:150]: ?
```js
onTouchTap={this.props.unmount}
```

**Review by rasendubi — COMMENTED** [review]
I would suggest trying to separate UI-related logic from another one.

Have you read [Thinking in React](https://facebook.github.io/react/docs/thinking-in-react.html)? It is an excellent article on how to design with React.

If I would follow the principle, I would design a modal dialog that takes articles in and returns an array of selected articles out when submit button is pressed. In that way, it could be completely reused for both import and export as all the processing would be factored out of the component.

File: src/components/ExportModal/ExportModal.js (hunk 2 above)
Comment (drets) [reply src/components/ExportModal/ExportModal.js:150]: `Handler function for onTouchTap prop key must begin with 'handle' [react/jsx-handler-names]`
(of course I may rename `unmount` to `handleUnmount`)

Comment (drets) [inline src/components/ExportModal/ExportModal.js]: (of course I can rename `unmount` to `handleUnmount` but it's really nitpicking)

**Comment by drets** [comment]
>Have you read Thinking in React?

Yes, today.

>I would design a modal dialog that takes articles in and returns an array of selected articles out when submit button is pressed. 

There are subtle differences between modals which kinda hardly to abstract:
• import modal may show an error if user chose file with invalid json
• by pressing on submit for importing I want to show a spinner since it makes a request to the server

Of course I could move showing error out from modal to another component and I could show the spinner not inside submit button but somewhere else (e.g. on top of the page), but I hardly believe that I'll gain something doing so, I am fine to have 2 modal components with a bit logic inside.

**Comment by rasendubi** [comment]
> There are subtle differences between modals which kinda hardly to abstract:
• import modal may show an error if user chose file with invalid json
• by pressing on submit for importing I want to show a spinner since it makes a request to the server

I would move them out of the dialog. These shouldn't belong to the dialog with the table anyway.

If an invalid file is selected for import, we shouldn't allow proceeding with the table.

When sending a request, close the table and show a transient dialog with the spinner. Close it as soon as the request completes. (What is the point of still showing the table when the user already submitted his articles?)

**Comment by drets** [comment]
>I would move them out of the dialog. These shouldn't belong to the dialog with the table anyway.

The error message is inside dialog (there is no table in this case)

>What is the point of still showing the table when the user already submitted his articles?

User may select a different article (which suddenly came to her mind) and import it (without re-importing whole file again.)

---
As I said I don't see huge benefit from abstracting it and moving these parts, if you want to refactor it − feel free, I better start implementing a new feature.

**Pull request merged**